src/Components/Forms/GeneralComponents/SaveForm/index.js

```diff
@@ -8,63 +8,84 @@ import { store } from 'index';
 import { FHIR } from 'Utils/Services/FHIR';
 import { useHistory } from 'react-router-dom';
 import { baseRoutePath } from 'Utils/Helpers/baseRoutePath';
+import Grid from '@material-ui/core/Grid';
 
 /**
  * @author Idan Gigi idangi@matrix.co.il
  * @param { [{value: string, label: string}] } statuses
  * @description The data from this component will be inside the data object as 'nextStatus' in the onSubmitHandler
  */
-const SaveForm = ({ statuses, encounter, onSubmit, validationFunction }) => {
+const SaveForm = ({
+  statuses,
+  encounter,
+  onSubmit,
+  validationFunction,
+  updateEncounterExtension,
+}) => {
   const { t } = useTranslation();
   const { permission, watch, getValues } = useFormContext();
 
   const history = useHistory();
 
   const selectedStatus = watch('nextStatus');
+  const practitioner = store.getState().login.userID;
+  const updateEncounter = !updateEncounterExtension
+    ? async () => {
+        try {
+          const cloneEncounter = { ...encounter };
+          cloneEncounter.extensionSecondaryStatus = selectedStatus;
+          cloneEncounter.status = 'in-progress';
+          cloneEncounter.practitioner = practitioner;
 
-  const updateEncounter = async () => {
-    try {
-      const cloneEncounter = { ...encounter };
-      cloneEncounter.extensionSecondaryStatus = selectedStatus;
-      cloneEncounter.status = 'in-progress';
-      cloneEncounter.practitioner = store.getState().login.userID;
+          await FHIR('Encounter', 'doWork', {
+            functionName: 'updateEncounter',
+            functionParams: {
+              encounterId: encounter.id,
+              encounter: cloneEncounter,
+            },
+          });
+          history.push(`${baseRoutePath()}/generic/patientTracking`);
+        } catch (error) {
+          console.log(error);
+        }
+      }
+    : async () =>
+        updateEncounterExtension(encounter, watch('nextStatus'), practitioner);
```

Comment:
My suggest is that updateEncounterExtension will return only the "cloneEncounter" and the logic of the update encounter and goto patient traking will be common.
LIke this:
```
const cloneEncounter = { ...encounter };
 const updateEncounter =  async () => {
 !updateEncounterExtension
    ?
          cloneEncounter.extensionSecondaryStatus = selectedStatus;
          cloneEncounter.status = 'in-progress';
          cloneEncounter.practitioner = practitioner;
     : cloneEncounter =  updateEncounterExtension(encounter, watch('nextStatus'), practitioner);


 try {
          await FHIR('Encounter', 'doWork', {
            functionName: 'updateEncounter',
            functionParams: {
              encounterId: encounter.id,
              encounter: cloneEncounter,
            },
    });
    history.push(`${baseRoutePath()}/generic/patientTracking`);
        } catch (error) {
          console.log(error);
        }
      }
  ```

Reply:
Sorry but I don't quite understand what is more needed here since this is what we talked about in the meeting you me and gigi....